pkg/controller/deadmanssnitch/deadmanssnitch_controller.go

```diff
@@ -240,6 +240,28 @@ func (r *ReconcileDeadMansSnitch) Reconcile(request reconcile.Request) (reconcil
 			}
 		}
 
+		// Get the snitch again to check status
+		snitches, err = r.dmsclient.FindSnitchesByName(snitchName)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		if len(snitches) > 0 {
```

Comment:
This block and the block starting at 226 are executing on the same conditions. It might make the code easier to read if they where combined and their actions abstracted into descriptive function calls, doTheThing() doTheOtherThing(). A suggestion, not a a requirement.

Reply:
Here again, we are now checking the return of the second snitch fetch. We could of comibined all of this be fetching the snitch before that first if loop, but that would make the api calls everytime a reconcile loop is called. This limits the number of times we hit the API since it has to pass that first if statement here https://github.com/openshift/deadmanssnitch-operator/blob/190ec5c31df9b2e96a292ffdc2fa6eff884102c6/pkg/controller/deadmanssnitch/deadmanssnitch_controller.go#L215